R/dependencies.R

```diff
@@ -146,18 +143,35 @@ yamlDeps <- function(yaml) {
 
 fileDependencies.Rmd <- function(file) {
 
+  # All Rmd files get an "rmarkdown" dependency.
+  deps <- c("rmarkdown")
+  
   # We need to check for and parse YAML frontmatter if necessary
-  yamlDeps <- NULL
   content <- readLines(file)
   if (hasYamlFrontMatter(content)) {
     tripleDashes <- grep("^---\\s*$", content, perl = TRUE)
     start <- tripleDashes[[1]]
     end <- tripleDashes[[2]]
-    yamlDeps <- yamlDeps(content[(start + 1):(end - 1)])
+    yaml <- content[(start + 1):(end - 1)]
+    yaml <- paste(yaml, collapse = "\n")
+    deps <- append(deps, yamlDeps(yaml))
+
+    if (requireNamespace("knitr", quietly = TRUE)) {
+      if (packageVersion("knitr") >= "1.10.18") {
+        knit_params <- knitr::knit_params_yaml(yaml, evaluate = FALSE)
+        if (length(knit_params) > 0) {
+          # Parameterized R Markdown documents can use a Shiny customization app.
+          deps <- append(deps, "shiny")
```

Comment:
Does this imply that any `.Rmd` containing custom parameters has `shiny` as an implicit dependency?